File: specs/NativeLoggerModule.ts
import {TurboModule, TurboModuleRegistry} from 'react-native';

export interface Spec extends TurboModule {
  registerLog(callback: (log: string) => void): void;
}

export default TurboModuleRegistry.getEnforcing<Spec>(
  'NativeLoggerModule',
);

File: shared/NativeLoggerModule.cpp
#include "NativeLoggerModule.h"

#include "LudoClient.h"

namespace facebook::react {

NativeLoggerModule::NativeLoggerModule(std::shared_ptr<CallInvoker> jsInvoker) 
        : NativeLoggerModuleCxxSpec(std::move(jsInvoker)) {}

void NativeLoggerModule::registerLog(jsi::Runtime& rt, jsi::Function callback) {
    callback_ = std::make_shared<jsi::Function>(std::move(callback));
    LudoClient::create(this);
}

void NativeLoggerModule::log(jsi::Runtime& rt, std::string message) {
    callback_->call(rt, jsi::String::createFromUtf8(rt, message));
}

}